ESRF/ESRFMultiCollect.py

```diff
@@ -328,11 +331,50 @@ def init(self):
     def take_crystal_snapshots(self, number_of_snapshots):
         self.bl_control.diffractometer.takeSnapshots(number_of_snapshots, wait=True)
 
-    #TODO: remove this hook!!!
     @task
     def data_collection_hook(self, data_collect_parameters):
-        return
+        # Metadata
+        try:
+            fileinfo =  data_collect_parameters["fileinfo"]
+            directory = fileinfo["directory"]
+            prefix = fileinfo["prefix"]
+            run_number = int(fileinfo["run_number"])
+            # Connect to ICAT metadata database
+            listDirectory = directory.split(os.sep)
+            beamline = "unknown"
+            proposal = "unknown"
+            if listDirectory[1] == "data":
+                if listDirectory[2] == "visitor":
+                    beamline = listDirectory[4]
+                    proposal = listDirectory[3]
+                else:
+                    beamline = listDirectory[2]
+                    proposal = listDirectory[4]
+            metadataManagerName = 'id30a1/metadata/ingest'
```

Comment:
As you said in your email Olof the Tango device names have to be put in xml file